src/lmake_func.cc

```diff
@@ -309,18 +316,20 @@ namespace lmake { namespace func {
         // Checks if there are any files on the corresponding directory, if not
         // returns with an error
         std::string result;
-        auto files = os::list_dir(path);
-        if(files.empty()) {
-            spdlog::error("Path of regex does not exist.");
-            std::exit(1);
-        }
-
-        // Loops through the files and tests the regex
-        for(std::string file : files) {
-            if (std::regex_search(file, match, regex_obj)) {
-                result.append(file + " ");
+        try {
+            // Loops through the files and tests the regex
+            std::filesystem::directory_iterator files(path);
+            for (auto& file : files) {
+                const auto& file_string = file.path().string();
+                if (std::regex_search(file_string, match, regex_obj)) {
+                    result.append(file_string + " ");
+                }
             }
         }
+        catch (const std::exception &e) {
```

Comment:
Does not follow the conventions used across the source code. Should be `const std::exception& e` instead of `const std::exception &e`.